applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -3,8 +3,116 @@
+      Bluetooth is configured to use Nordic's proprietary link-layer.
+      This is to achieve gaming grade performance.
+      The preconfigured build types configure the device with or without the bootloader and in debug or release mode.
+    * nrf52840_pca10059 - dongle.
+      The configuration uses Nordic's nRF52840 dongle defined in Zephyr.
+      Since the board is generic a project specific changes are applied in dts overlay file.
+      The application is configured to act as both a mouse and a keyboard.
+      Bluetooth uses Nordic's proprietary link-layer and is configured to act as a central.
+      Input data come from Bluetooth and are retransmitted to the USB.
+      The preconfigured build types configure the device with or without the bootloader and in debug or release mode.
+    * nrf52_pca20037 - keyboard.
+      The board used is defined in ``nrf/boards/arm/nrf52_pca20037`` for project specific HW.
+      The application is configured to act as a keyboard with both BLE and USB transports enabled as input.
+      Bluetooth is configured to use Nordic's proprietary link-layer.
+      The preconfigured build types configure the device with or without the bootloader and in debug or release mode.
+    * nrf52_pca20044 and nrf52810_pca20045 - desktop mice.
+      Both boards are for project specific HW and located in ``nrf/boards/arm/nrf52_pca20044`` and ``nrf/boards/arm/nrf52810_pca20045`` respectively.
+      The application is configured to act as a mouse.
+      Only the BLE transport is enabled.
+      The bluetooth uses Zephyr's SW link-layer.
+      For the nrf52810 device there is no configuration with bootloader available.
+    * nrf52840_pca10056 - sample mouse or keyboard
+      The configuration uses the widely available nrf52840 DK.
+      The build types allow to build mouse both as mouse and keyboard.
+      Inputs are simulated.
+      The configuration with bootloader is available.
+
+Adding a new board
+------------------
+      
+When creating your own board start by defining a board copying board files that is the closest match for your intended HW.
+For example if the goal is to create a gaming mouse ``nrf/boards/arm/nrf52840_pca20041`` should be the base.
+
+After that the dts files should be edited to make sure HW configuration is reflected. Pay attention on pins being used and bus configuration for optical sensor.
+
+The next step is preparing the application configuration files.
+Again it is recommented to start by copying the project files for the device that is the closest match for your intended HW.
+In case of a gaming mouse this would be files located in ``applications/nrf_desktop/configure/nrf52840_pca20040``.
+Depending on a chosen board you may need to edit dts overlay file.
+If you have crated a new board and its dts files fully describe your HW this will not be needed and overlay file may be empty.
+
+At the beginning focus should be put on a single configuration first.
```

Comment:
`At the beginning focus should be put on a single configuration`?

Reply:
ok